.github/workflows/csv-coverage-pr-artifacts.yml

```diff
@@ -89,9 +89,25 @@ jobs:
       - name: Save PR number
         run: |
           mkdir -p pr
-          echo ${{ github.event.pull_request.number }} > pr/NR
+          echo ${PR_NUMBER} > pr/NR
+        env:
+          PR_NUMBER: ${{ github.event.pull_request.number }}
       - name: Upload PR number
         uses: actions/upload-artifact@v3
         with:
           name: pr
           path: pr/
+      - name: Save comment ID (if it exists)
+        run: |
+          mkdir -p comment
+          # Find the latest comment starting with COMMENT_PREFIX
+          COMMENT_PREFIX=":warning: The head of this PR and the base branch were compared for differences in the framework coverage reports."
+          gh api "repos/${GITHUB_REPOSITORY}/issues/${PR_NUMBER}/comments" --paginate | jq --arg prefix "${COMMENT_PREFIX}" '[.[] | select(.body|startswith($prefix)) | .id] | max' > comment/ID
```

Comment:
Minor: `.[]` in a filter query always confuses me, because I can't work out the type of the result (it's a newline separated string rather than an array). Consider using `map(select(...))` instead, then your result is still a filtered array, and you can pass that to `max`. If you take my suggestion in the file below, you can end with `// empty` so that if the array is empty, you'll get the empty string as the result.

Reply:
Updated in https://github.com/github/codeql/pull/14517/commits/687ecffe71acf17ca8901f434d2fff854ab2788d.